crates/bevy_pbr/src/render/mesh_bindings.rs

```diff
@@ -17,30 +17,18 @@ mod layout_entry {
     use crate::MeshUniform;
     use bevy_render::{
         render_resource::{
-            BindGroupLayoutEntry, BindingType, BufferBindingType, BufferSize, GpuArrayBuffer,
-            ShaderStages, TextureSampleType, TextureViewDimension,
+            binding_types::uniform_buffer, BindGroupLayoutEntry, BindingType, BufferSize,
+            GpuArrayBuffer, ShaderStages, TextureSampleType, TextureViewDimension,
         },
         renderer::RenderDevice,
     };
 
     fn buffer(binding: u32, size: u64, visibility: ShaderStages) -> BindGroupLayoutEntry {
-        BindGroupLayoutEntry {
-            binding,
-            visibility,
-            count: None,
-            ty: BindingType::Buffer {
-                ty: BufferBindingType::Uniform,
-                has_dynamic_offset: true,
-                min_binding_size: BufferSize::new(size),
-            },
-        }
+        uniform_buffer(true, BufferSize::new(size)).build(binding, visibility)
```

Comment:
this function feels pretty useless now, it should be flattened out. happy to defer that if you prefer though, there is probably a fair bit more that could be cleaned up here.

Reply:
Yeah, I have a branch where I made more changes to this part of the code but the diff started getting pretty messy. I can push it anyway I guess. I was considering making a follow up PR for this.

Reply:
pushed the change. It's a bit better, but it also has a similar issue that setting visibility is a bit weird.